### PR TITLE
Merge pull request #40 + #58

### PR DESCRIPTION
**Pull request by joswinter**
#12 Calculate relative coordinates of nodes in DNA sequence graph
#13 Insert empty nodes (insertion / deletion) in DNA sequence graph
#15 Indicate mutation types based on empty tiles and relative

coordinates (insertion, deletion and SNP)
#54 Improve Graph overview: Scaling of blocks etc.

### CODE REVIEW

**Comment by jorenham** [comment]
Code mostly looks good, couple of remarks:
- Commit `df7fdca` is way too big, you should split it up (sorry).
- In `AlignmentTraverser.bridgeSequence` you should use `Set` instead of `HashSet` as method argument. Also the method body is hard to read. (bad readability is an issue in more methods, don't be afraid to use variables to improve readability)
- For all mutations, you've set the color inside the class itself. This is not very MVC; you should put these in css and add a styleclass to the element for the mutation, another way would be to make it a view instead of a model.
- In the javadocs, you often put a `-` in front of a parameter description, this does not comform to the javadoc standards.
- In `VertexView` there is alot of styling, this should be moved to css.
- The package structure of the tests are not parallel with the structure of the code itself.

**Comment by joswinter** [comment]
The commits are actually split up as the title states:
- Pull Request #40 
- Pull Request #58 

In Pull Request #58 I already discussed a few points regarding the styling. We should alter styling from the whole graph into the stylesheet, I created an issue for it #60.

The "-" were introduced by Arjan in Pull Request #46.

**Comment by albert-smit** [comment]
javadoc comments in TileView contain trailing spaces

**Comment by rutgervandenberg** [comment]
##### TLDR: This is a mess. I don't agree with merging it at this point. Too tired to take a proper look at it right now. Some initial remarks:
- Please never ever ever ever make a mega-commit like this again. 
  - You didn't rebase, so you're actually reverting a lot of previous commits
  - It is unclear which part of the code accomplishes what, which is usually made clear by smaller, meaningful commits. 
  - Because you're undoing a bunch of old commits it is very hard to find out what you actually changed.
- Please run the code formatter. 
- Rebase.
- Please remove @SuppressWarnings eclipse annotations. 
- AbsoluteProjectionTraverser.traverseVertexBackward(): destination = getsource. Rename this.
- AlignmentTraverser.setCopy() is useless. Copying a set is a single statement anyways, no need to make a method for it. 
- AlignmentTraverser: SequenceSegments are related to vertices not edges. 
- Interface GraphTraverser is useless. It defines a single method but does not tell us what it is meant to accomplish, only that it traverses the graph. Which is implementation, and should not be defined in the interface. Similarly implementations of this interface have meaningless method names, which only serves to confuse people.
- Rebase.
- Interface SegmentContent, the length method (should be getLength at the very least) is redundant. Length is implicit to the string you're already retrieving. No need to explicitly store it. 
- Interface Mutation: Color should not be stored in code, but in styling. When this is removed the entire interface is empty and thus useless. Even if the color is not removed, having an interface and a bunch of implementing classes _just_ for storing colors is bad.
- TileView.checkAvailable(): 
  - checking for <code>i >= lanes.size()</code> on each iteration is pointless. It is only ever going to be true once, namely on the last iteration. On every other iteration you're wasting time.
  - this method does not just check availability but it also draws the vertex? Seems to need a rename.
- Rebase.
- TileView.segmentFree(): argument <code>i</code> should have a proper name. 
- TileView.segmentInsert(): argument <code>i</code> should have a proper name. 
- TileView: The list <code>lanes</code> is passed around by a bunch of private methods. This is confusing, you should just change its scope to the entire class.
- VertexView: Does font size and tile spacing vary between tiles(weird)? If not, these should be static. 
- GraphAdapterTest: assert should be assertTrue

**Comment by joswinter** [comment]
Thanks for the comments!
- I formatted the code checkstyle-friendly, I think I solved almost all the checkstyle problems.
- I already removed the @SuppressWarnings eclipse annotations
- Fixed traverseVertexBackward
- Removed setCopy()
- changed length into getLength
- getLength method isn't redundant with regards to SegmentEmpty which does contains just a length value. 
- I discussed with Joren that we will eventually want to store other information in mutations for example if we want to be able to express more advanced mutation types such as translocation and inversions. I agree that currently it looks like overkill.
- I also discussed with Joren that we will eventually want to use the traversers as decorators if we are going to combine everything.
- I changed a lot of checkstyle and javadoc problems.
- Implemented the loading of parsed data set into TileView, it works very fast.

**Comment by joswinter** [comment]
Removed traverser code for a future pull request. Traversing system needs a redo because of the 'actual' coordinate system. I think I know what needs to be done to achieve that, I will reopen the issue.

**Comment by jorenham** [comment]
Look good now.

**Comment by albert-smit** [comment]
code looks good but has a few minor issues
- javadoc of sequenceSeqment is missing a newline, and has basically empty javadoc comments
- javadoc comments in edgeLine still contain `-`
- sequenceSegmentTest is missing a newline at end of file
- its preferable to not change the .gitignore to add files that are ignored

**Comment by alangerak** [comment]
no problems found, looks good

**Comment by rutgervandenberg** [comment]
- JGraphTGraphAdapter: 
  - Does not end with a newline
  - missing class javadoc
- DefaultSequence:
  - Has equals but not hashcode. This is especially problematic as we do put sequences in hashmaps.
- SequenceSegment
  - Casts primitive long to long object. Should just use Long.valueOf()
  - has compareTo but not equals/hashcode. This can cause strange behaviour. 
  - should rename the accessor methods for 'absolute' coordinates
- SegmentContent
  - getLength is redundant since it is just metadata of the content string. You can store it in the case of EmptySegment, but since you do return strings of an appropriate length in toString you don't need the accessor method for it. Better yet would be to just create the "empty" string in the constructor and not explicitly store the length.
- Mutation: 
  - Should probably just hold off on adding this stuff until you can actually use (and thus test) it.

File: src/main/java/nl/tudelft/lifetiles/graph/controller/GraphController.java [hunk 1]
```diff
@@ -22,8 +30,28 @@
...
+        Group root = controller.drawGraph();
+        wrapper.setContent(root);
+    }
+
+    /**
+     * Loads a graph by filename.
+     *
+     * @return parsed graph by filename.
+     */
+    private Graph<SequenceSegment> loadGraphModel() {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/controller/GraphController.java:48]: I assume this is an obsolete method? If that's the case you should mark it with `@Deprecated`.

File: src/main/java/nl/tudelft/lifetiles/graph/controller/TileController.java [hunk 2]
```diff
@@ -65,15 +65,15 @@ public final Group drawGraph() {
...
      * @param args
-     *            - Arguments for the function that needs to be called.
+     *            Arguments for the function that needs to be called.
      */
     public final void setChangedProperty(final String propertyName,
             final List<Object> args) {
 
         if (propertyName.equals("COLOR_CHANGED")) {
-            viewVar.changeVertexColour((Vertex) args.get(0),
+            viewVar.changeVertexColour((VertexView) args.get(0),
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/controller/TileController.java:76]: The casting is prone to throw throw an error if the class' post-condition (first argument a `VertexView`, second one a `Color`) is not met. You should check this sooner re-throw it as a `IllegalArgumentException` with an appropriate message.

File: src/main/java/nl/tudelft/lifetiles/graph/models/jgrapht/JGraphTGraphAdapter.java [hunk 3]
```diff
@@ -157,13 +158,20 @@ public final V getDestination(final Edge<V> e) {
     }
 
     /**
+     * @return All vertices that have no outgoing edges.
+     */
+    @Override
+    public Set<V> getSink() {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/models/jgrapht/JGraphTGraphAdapter.java:164]: Should be `getSinks()`.

File: src/main/java/nl/tudelft/lifetiles/graph/models/jgrapht/JGraphTGraphAdapter.java [hunk 4]
```diff
@@ -182,4 +190,28 @@ private DefaultEdge unpackEdge(final Edge<V> input) {
...
+    /**
+     * Divides an edge into two edges with an inserted vertex in the middle.
+     * 
+     * @param edge
+     *            Edge to be divided.
+     * @param vertex
+     *            Vertex to be inserted.
+     */
+    @Override
+    public void divideEdge(Edge<V> edge, V vertex) {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/models/jgrapht/JGraphTGraphAdapter.java:202]: `splitEdge()` might be more suitable.

File: src/main/java/nl/tudelft/lifetiles/graph/controller/TileController.java (hunk 2 above)
Comment (alangerak) [inline src/main/java/nl/tudelft/lifetiles/graph/controller/TileController.java:76]: This function together with changeColour can be removed. I added this code in advance for filtering but I have altered it a lot.
